cips/cip-shwap_protocol.md

```diff
@@ -0,0 +1,407 @@
+---
+cip: 18
+title: Shwap Protocol 
+description: Shwap - a new messaging framework for DA and sampling
+author: Hlib Kanunnikov (@Wondertan)
+discussions-to: https://forum.celestia.org/t/cip-shwap-protocol/1551
+status: Draft
+type: Standards Track
+category: Data Availability, Networking
+created: 2024-02-02
+---
+
+## Abstract
+
+This document specifies Shwap - the simple and expressive yet extensible messaging framework aiming to
+solve critical inefficiencies and standardize messaging of Celestia's Data Availability p2p network.
+
+Shwap defines a messaging framework to be exchanged around the DA p2p network in a trust-minimized way without enforcing
+transport (QUIC/TCP or IP) or application layer protocol semantics (e.g., HTTP/x). Using this framework, Shwap
+declares the most common messages and provides options for stacking them with lower-level protocols.
+Shwap can be stacked together with application protocol like HTTP/x, [KadDHT][kaddht], [Bitswap][bitswap] or any custom
+protocol.
+
+## Motivation
+
+The current Data Availability Sampling (DAS) network protocol is inefficient. A _single_ sample operation takes log₂(k)
+network roundtrips (where k is the square size). This is not practical and does not scale for the theoretically unlimited
+data square that the Celestia network enables. The main motive here is a protocol with O(1) roundtrip for _multiple_
```

Comment:
[nit] does Celestia theoretically enable an unlimited data square? I haven't seen that idea expressed before and the whitepaper talks about how scaling the square size up depends on a sufficient number of light nodes sampling. If Celestia enables an unlimited data square then we'd need an unlimited number of light nodes.

IMO this proposal would be more convincing if it stated something like:

```
This is not practical and does not scale as the Celestia square size is increased.
```

Reply:
Yes, having enough light nodes is needed, but it doesn't contradict the fact that it is theoretically unlimited. Having enough light nodes is a requirement, but not a theoretical limit.